app/controllers/nwbib/Application.java

```diff
@@ -995,17 +995,23 @@ private static Promise<Result> deleteFromIndex(String id) throws UnsupportedEnco
 
 	private static Promise<Result> transformAndIndex(String id, JsonNode jsonBody)
 			throws IOException, FileNotFoundException, RecognitionException, UnsupportedEncodingException {
-		JsonNode transformedJson = transform(jsonBody);
-		Promise<JsonNode> dataPromise = id.startsWith("f") && transformedJson.has("hbzId") ? // hbz-Fremddaten
-				addToLobidData(transformedJson) : Promise.pure(transformedJson);
+		JsonNode transformedJson = transformStrapiToLobid(jsonBody);
+		Promise<JsonNode> dataPromise = Promise.pure(transformedJson);
+		if (id.startsWith("f")) { // Fremddaten
+			if (transformedJson.has("hbzId")) {
+				dataPromise = addToLobidData(transformedJson);
+			} else if (transformedJson.has("hebisId")) {
+				dataPromise = addToHebisData(transformedJson);
+			}
+		}
 		return dataPromise.flatMap(result -> {
 			Cache.remove(String.format("/%s", id));
 			WSRequest request = WS.url(elasticsearchUrl(id)).setHeader("Content-Type", "application/json");
 			return request.put(result).map(response -> status(response.getStatus(), response.getBody()));
 		});
 	}
 
-	private static JsonNode transform(JsonNode jsonBody)
+	private static JsonNode transformStrapiToLobid(JsonNode jsonBody)
 			throws IOException, FileNotFoundException, RecognitionException {
 		File input = new File("conf/output/test-output-strapi.json");
 		File output = new File("conf/output/test-output-0.json");
```

Comment:
Not so important, but I stumbled over this "test" because iMO it implies test like junittests (i.e. vs "production"). However what you unambigously mean is "validate", yes?
(same for `rpb-test-titel-to-lobid.flux` ).

Reply:
Yes, true, it's confusing, thanks. Renamed the files in https://github.com/hbz/rpb/pull/105/commits/d174dcea66419410acd5576ff12ce1a8dfe43cb7.